Devs-AA/Barefoot
# #168662105 Request approval or rejection functionality

#### What does this PR do?
  Enables managers to reject or approve request made by direct staff.

#### Description of Task to be completed?
  - Ensure only users with a manager role can access this endpoint.
  - Ensure managers can reject a request made only by their direct staff.
  - Ensure managers can approve a request made only by their direct staff..

#### How should this be manually tested?
  - git clone this repo using the command `git@github.com:Devs-AA/Barefoot.git`
  - cd to the project
  - checkout to the branch using the command `ft-request-approval`
  - create a `.env` file and set PORT,JWT_SECRET and DB variables as specified in the .env.example file.
  - Run the command `npm install` in the terminal.
  - Open a terminal and run `npm test`.
  - Open a terminal and run `npm run migrate:dev` and `npm run seed:dev`
  1. Open Postman, make a post request to `localhost:<port>/api/v1/users/auth/login`, with any of the below in the request body.
- `{email: requester2@gmail.com, password: Password1$}`
2. Copy the token in the reponse
3. Make a POST request to `http:localhost:<port>/api/v1/requests` with the token obtained above in the request header as `authorization`=`Bearer <token obtained>`
 4. Send a request object containing the following for a one-way trip:
         `{
           reason: 'Business',
           departmentId: 2,
           tripType: 'oneWay',
        }`

5. Make a post request to `localhost:<port>/api/v1/users/auth/login`, with the below  in the request body.
 `{email: barefoot2@gmail.com, password: Password1$}`
6. Make a PATCH request to `http:localhost:<port>/api/v1/requests<requestId>` with the request body containing. Note that status can only be approved or rejected
          `{
            "status": "approved"
           }`
 

7. Repeat steps 5 & 6 with the object 
`{
  email: barefoot1@gmail.com, password: Password1$
}`
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[##168662104](https://www.pivotaltracker.com/story/show/#168662104)
[##168662105](https://www.pivotaltracker.com/story/show/#168662105)

#### Screenshots (if appropriate)
![Screenshot (81)](https://user-images.githubusercontent.com/46751363/65627123-9d794580-dfc6-11e9-9e31-f65b359e29ed.png)
![Screenshot (82)](https://user-images.githubusercontent.com/46751363/65627134-a10ccc80-dfc6-11e9-9515-c6cb24b1038b.png)
![Screenshot (83)](https://user-images.githubusercontent.com/46751363/65627354-1c6e7e00-dfc7-11e9-985d-aecd9bc25a78.png)



#### Questions:
N/A
![Screenshot (81)](https://user-images.githubusercontent.com/46751363/65627107-95210a80-dfc6-11e9-8d82-a65d9aaf2a5e.png)
